### PR TITLE
Feature/user list

### PR DESCRIPTION
**Pull request by keith1111**
address :    /users  - show all users   (UserList.js)
/users/  user_id    - show full info of user with that id     (UserPage.js)

Maybe we should add field 'login' to database?
in that case link to user page will be like: /users/myLogin    
and now it is like  /users/wefhwufh238yr8923yrghfigriweg
i can not prefer field 'name' for these purposes because names can be duplicate (John Smith, Xing Lee, Maria Fernandez)

### CODE REVIEW

**Comment by nikrb** [comment]
the user is available in the backend as req.user, after login. We're planning to use github for main login so unique name won't be an issue, for now atleast. so route `/user/:id*?` will return detail of specified user id, or logged in user if param is not specified. Always open to discussion, but that's my initial take

**Comment by nikrb** [comment]
Looks great, but before we merge I'd like to hear @giovannisampaio and @jenovs comments on the following:
1. I think clicking on the users menu should show all users.
2. We need another menu item 'profile' to show my profile. (Not suggesting we do this in this pr, just throwing it out there)
3. Why open a new tab when we click on a user to see that user's profile?

**Comment by keith1111** [comment]
OK lets listen and i will rewrite if we decide to change something

File: frontend/src/App.js [hunk 1]
```diff
@@ -74,7 +79,7 @@ class App extends Component {
               path="/assets/:id"
               render={() => <AssetEdit {...assetData} />}
             />
-            <Route path="/users" render={() => <div>Users component</div>} />
+
```
Comment (jenovs) [inline frontend/src/App.js]: Unnecessary line

File: frontend/src/UserList/UserList.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+              <th>Name</th>
+              <th>Role</th>
+            </tr>
+            {mockUsers.map(d => (
+              <tr key={d._id}>
+                <td>
+                  <img className="avatar" src={d.avatar} alt="avatar" />
+                </td>
+                <td>
+                  <a target="_blank" href={`/users/${d._id}`}>
```
Comment (jenovs) [inline frontend/src/UserList/UserList.js]: Use `Link` and `Route` from `react-router-dom` to create navigation inside the same page.

**Review by jenovs — CHANGES_REQUESTED** [review]
Replace opening of new page with router navigation.

**Comment by jenovs** [comment]
Regarding @nikrb comment:
1. I suppose it should be admin only feature, so, yes `Users` should show list of all users.
3. The whole idea behind single page app is that everything happens on a... single page :grinning: 

Apart from that it's fine.

P.S. I personally don't like creating tables manually (I did it for `Articles` component), so for `Assets` I used external component. I recommend to look into it.

**Comment by keith1111** [comment]
Replaced tag a with link
Rewrote href of links to lead them to wished addresses
Added some exception to eslint config (to remove odd errors)

File: frontend/src/UserList/UserPage.js [hunk 3]
```diff
@@ -53,7 +55,12 @@ export default class UserPage extends React.Component {
       );
     }
 
-    return <section className="user-page">{userInfo}</section>;
+    return (
+    <section className="user-page">
+	    {userInfo}
```
Comment (nikrb) [inline frontend/src/UserList/UserPage.js]: code indents look wrong here but look ok in local editor, weird

File: .eslintrc.json [hunk 4]
```diff
@@ -13,6 +13,13 @@
     "react/forbid-prop-types": "off",
     "react/jsx-filename-extension": "off",
     "react/no-danger": "off",
-    "react/prefer-stateless-function": "off"
+    "react/prefer-stateless-function": "off",
+    "jsx-a11y/anchor-is-valid": [ "error", {
```
Comment (nikrb) [inline .eslintrc.json:17]: not sure about this. @jenovs can you have a look at this please. linter is complaining Link tag requires an href attribute, which doesn't make sense to me.

**Comment by nikrb** [comment]
sorry keith we have ~three~ two failing tests- `npm test` in frontend directory. Shout it you need a hand.
1. src/Navbar/Navbar.test.js
2. ~src/AssetEdit/utils.test.js~
3. src/LandingPage/LandingPage.test.js

~@jenovs the second one, do you see the same?:~
```
    /home/nik/projects/voyage4/bears-25/frontend/src/AssetEdit/utils.test.js:10
      it('readFile should convert file to dataURL', async () => {
```
~That doesn't look right, it's complaining about async. Am I doing something wrong?~
Point 2 isn't a problem, I was running wrong version of node, doh!

**Comment by nikrb** [comment]
@keith1111 Point number 3 is a generated css class change. Landing page still looks good to me so easy to fix, just update the snapshot, `npm test -- -u`.

**Comment by jenovs** [comment]
1. A link was added to `Navbar` so test should be changed to expect `7`
3. Landing page is missing `import 'jest-styled-components';`

**Comment by nikrb** [comment]
I have added `jest-styled-components` to issues #58

File: .eslintrc.json [hunk 5]
```diff
@@ -13,6 +13,13 @@
...
+            "components": [ "Link" ],
+            "specialLink": [ "to" ]
+            
+          }]
+    
+
```
Comment (jenovs) [inline .eslintrc.json]: Unnecessary whitespace

**Comment by keith1111** [comment]
removed whitespace

**Comment by nikrb** [comment]
@keith1111 `jest-styled-components` output looks good, but you need to update the snapshots in order for the tests to pass :)

Also I still see linter errors:
```
/home/nik/projects/voyage4/bears-25/frontend/src/LandingPage/LandingPage.test.js
  4:8  error  Absolute imports should come before relative imports  import/first

/home/nik/projects/voyage4/bears-25/frontend/src/UserList/UserList.js
  3:1  error  Expected empty line after import statement not followed by another import  import/newline-after-import

/home/nik/projects/voyage4/bears-25/frontend/src/UserList/UserPage.js
  4:1  error  Expected empty line after import statement not followed by another import  import/newline-after-import
```

**Review by jenovs — APPROVED** [review]

**Review by jenovs — APPROVED** [review]

**Review by giovannisampaio — APPROVED** [review]

**Pull request merged**